app/configurator/components/ui-helpers.ts

```diff
@@ -91,16 +108,74 @@ const getFormattersForLocale = memoize((locale) => {
     minute: format("%d.%m.%Y %H:%M"),
     hour: format("%d.%m.%Y %H:%M"),
     day: format("%d.%m.%Y"),
-    month: format("%d.%m.%Y"),
+    month: format("%m.%Y"),
     year: format("%Y"),
   };
 });
 
-const useLocalFormatters = () => {
+export const useLocalFormatters = () => {
   const locale = useLocale();
   return getFormattersForLocale(locale);
 };
 
+const dateFormatterFromDimension = (
+  dim: TemporalDimension,
+  localFormatters: LocalDateFormatters,
+  formatDateAuto: (d: Date | string | null) => string
+) => {
+  if (
+    dim.timeFormat &&
+    dim.timeUnit &&
+    localFormatters[dim.timeUnit.toLowerCase() as keyof typeof localFormatters]
+  ) {
```

Comment:
After taking a second look at this, do we even use timeFormat defined in CC somewhere? Here it looks we only check if it was defined, but is it taken into consideration when formatting the dates?

Reply:
Not that I know of, since we use d3 for parsing, we'd have to convert the parsing spec from CC format to d3 format. For now we only rely on the timeUnit. Let's see if we encounter cases where the timeFormat has a different time format than what's defined based on the timeUnit.

@l00mi we currently use the below code to know how to parse dates coming from a cube ? Can we rely on this ?

```
const parseSecond = timeParse("%Y-%m-%dT%H:%M:%S");
const parseMinute = timeParse("%Y-%m-%dT%H:%M");
const parseDay = timeParse("%Y-%m-%d");
const parseMonth = timeParse("%Y-%m");
const parseYear = timeParse("%Y");

const dateParsers = {
  Year: parseYear,
  Month: parseMonth,
  Day: parseDay,
  Minute: parseMinute,
  Second: parseSecond,
};

// When parsing observations
const dateParser = dateParsers[dimension.timeUnit]
```

Reply:
I wonder then why we throw a fatal error when timeFormat is not defined in CC, when we do not make use of it (I know it's not in the scope of this PR, but thinking out loud since the topics are connected). I think that either then we should take it into consideration (if @l00mi agrees) or remove the constraint of having non-empty timeFormat  – yesterday Veronique couldn't load a cube because of it, despite having other fields filled up correctly.

Reply:
(it's defined as mandatory field in GQL)

Reply:
Oh I made a mistake.

1. timeFormat is returned by us, from the resolvers, based on the data type of the dimension.

```
const timeFormats = new Map<string, string>([
  [ns.xsd.gYear.value, "%Y"],
  [ns.xsd.gYearMonth.value, "%Y-%m"],
  [ns.xsd.date.value, "%Y-%m-%d"],
  [ns.xsd.dateTime.value, "%Y-%m-%dT%H:%M:%S"],
]);
```

2. This way I do not need the conversion table and can directly us the timeFormat to build the d3 parser.

---

For Veronique's problem, it's not clear for me why the time format was not returned, based on this information @bprusinowski what do you think ?

Reply:
Cool – I also forgot it's done like that, I had the assumption it is fetched from CC 😆 I re-visited Veronique's problem and [it was related to missing dataKind](https://github.com/visualize-admin/visualization-tool/issues/498#issuecomment-1129790684) so all good from my side!

Reply:
Sorry, missed this, but yes it is all based on [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601). 

It should be as simple as `const date1 = new Date(isoStr1);` see also https://bobbyhadz.com/blog/javascript-convert-iso-string-to-date-object.

Reply:
Ah you mean the ticks, yes that is what we propose, but we can never be sure that this are the only once used by others. So best have a fallback to one unit in case it is not "known".
<img width="170" alt="Screenshot 2022-05-19 at 15 20 42" src="https://user-images.githubusercontent.com/502518/169303402-941af47d-ee1e-4703-a99d-dd42e638ef64.png">